catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/BaseSolrCatalogProvider.java

```diff
@@ -163,6 +185,15 @@ public boolean isAvailable(SourceMonitor callback) {
     return isAvailable(); // then trigger an active ping
   }
 
+  public boolean isAvailable(long timeout, long pollInterval, TimeUnit unit) {
+    try {
+      return solr.isAvailable(timeout, pollInterval, unit);
+    } catch (InterruptedException e) {
```

Comment:
❗️ Improper way to handle interruptions. They need to be propagated up by re-interrupting the current thread